community/security/src/main/java/org/neo4j/server/security/auth/UserSerialization.java

```diff
@@ -72,21 +72,28 @@
     private String serialize( User user )
     {
         return join( userSeparator, user.name(),
+                user.group(),
```

Comment:
Last time we discussed this you agreed that it would make sense to keep the Unixy way of a default group in the user file for convenience, and then add a group file. I am not sure what you mean by unclean intermediate state?